voici/addon.py

```diff
@@ -130,6 +130,32 @@ def post_build(self, manager):
                 ],
             )
 
+        # Update index page
+        yield dict(
+            name=f"voici:update_index:{self.manager.output_dir}",
+            actions=[
+                (
+                    self.update_index,
+                    [self.manager.output_dir / "voici"],
+                )
+            ],
+        )
+
+    def update_index(self, desc: Path):
+        """Update the redirect URL"""
+        contents_path = self.manager.contents[0]
+        if contents_path.is_file():
```

Comment:
IIRC `contents` can be a list of files, in this case you do want to generate a tree view.

So you may want to have a different check here?
```suggestion
        if len(self.manager.contents) == 1 and self.manager.contents[0].is_file():
```

Reply:
`contents` can be a list of files when used in another case than the CLI. For example we can very well use the Voici addon with jupyterlite-sphinx, which will not use the Voici CLI.